File: Python/#917_reverseOnlyLetters.py
class Solution:
    def reverseOnlyLetters(self, S: str) -> str:
        i=0
        j=len(S)-1
        s=list(S)
        while(i<j):
            if s[i].isalpha()==False:
                i+=1
            elif s[j].isalpha()==False:
                j-=1
            else:
                s[i],s[j]=s[j],s[i]
                i+=1
                j-=1
        return ''.join(s)
            
        

File: Java/#917_reverseOnlyLetters.java
class Solution {
    public String reverseOnlyLetters(String S) {
        char[] c = S.toCharArray();
        int i=0;
        int j=c.length-1;
        
        while(i<j){
            while(i<j && !Character.isLetter(c[i])) i++;
            while(j>i && !Character.isLetter(c[j])) j--;
            char t = c[i];
            c[i]=c[j];
            c[j]=t;
            i++;
            j--;
        }
        return new String(c);
    }
}
